### PR TITLE
Comments

### LINKED ISSUE #23: Add comment to a post

### LINKED ISSUE #24: View comments on a post

### PR DESCRIPTION
**Pull request by horvatmatthew**
Closes #23 and Closes #24 

Comments can be added to a post.  UI Created to add comment and show comments

### CODE REVIEW

File: FoxHound.Client/src/Comments/Comment.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    comment : CommentModel;
+}
+
+const Comment : React.FC < IProps > = (props) => {
+    const useStyles = makeStyles((theme) => ({
+        root: {
+            padding: theme.spacing(2, 3),
+            marginBottom: '10px'
+        }
+    }));
```
Comment (jjh121) [inline FoxHound.Client/src/Comments/Comment.tsx:16]: We've been using a mix of the the useStyle with a root class and using the "<Box"> component. We should standardize on how we want to do this at some point.

File: FoxHound.Client/src/Navigation/Content.tsx [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
-      <Route path="addBlog">
-        <AddBlog handleBlogAdded={(id: number) => {}} />
+      <Route path="/addBlog">
+        <AddBlog handleBlogAdded={(id : number) => {}}/>
       </Route>
-
-      <Route path="/blog/:blogId/addPost">
-        <AddPost />
+      <Route path="/blog /: blogId / addPost ">
+        < AddPost/>
```
Comment (jjh121) [inline FoxHound.Client/src/Navigation/Content.tsx:17]: Watch the spacing in the path here. That may break things.

File: FoxHound.Web/Controllers/CommentController.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FoxHound.Web.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
```
Comment (jjh121) [inline FoxHound.Web/Controllers/CommentController.cs:13]: This shouldn't be needed as the BaseApiController contains this to keep consistency across all controllers.

File: FoxHound.Web/Controllers/CommentController.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        public CommentController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        [HttpPost("[action]")]
+        public async Task<int> Create(CreateCommentCommand command)
+        {
+            int postId = await _mediator.Send(command);
+            return postId;
```
Comment (jjh121) [inline FoxHound.Web/Controllers/CommentController.cs:27]: I think postId should be commentId

File: FoxHound.App/Blogs/CreateComment/CreateCommentCommandHandler.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        public CreateCommentCommandHandler(IFoxHoundData foxHoundData)
+        {
+            _foxHoundData = foxHoundData;
+        }
+
+        public async Task<int> Handle(CreateCommentCommand request, CancellationToken cancellationToken)
+        {
+            Post post = await _foxHoundData.Posts
+                .Include(x => x.Comments)
+                .SingleAsync(x => x.PostId == request.PostId, cancellationToken);
```
Comment (jjh121) [inline FoxHound.App/Blogs/CreateComment/CreateCommentCommandHandler.cs:26]: If we keep Blog as the aggregate root, then this query would change.

File: FoxHound.App/FoxHound.App.csproj [hunk 6]
```diff
@@ -6,6 +6,10 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
+  <ItemGroup>
+    <None Remove="Blogs\CreateComment\CreateCommentCommandValidator.cs~RF1c45666.TMP" />
+  </ItemGroup>
```
[the remark this answers is not in the document]
Comment (horvatmatthew) [reply FoxHound.App/FoxHound.App.csproj:11]: Done.  I thought I had removed it

File: FoxHound.Client/src/Comments/AddComment.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+import Axios from "axios";
+import {useParams} from "react-router-dom";
+
+interface IProps {
+    postId : number
+}
+
+const AddComment : React.FC < IProps > = (props) => {
+    const [author,
+        setAuthor] = useState < string > ("");
```
[the remark this answers is not in the document]
Comment (horvatmatthew) [reply FoxHound.Client/src/Comments/AddComment.tsx:20]: Yes, I think while trying to add stuff in to test it messed up the formatting.  Once I resaved the file the formatting corrected itself.

File: FoxHound.Web/Controllers/CommentController.cs (hunk 3 above)
Comment (horvatmatthew) [reply FoxHound.Web/Controllers/CommentController.cs:13]: Removed.  I think I created the file as a controller which added those items in.

File: FoxHound.Web/Controllers/CommentController.cs (hunk 4 above)
Comment (horvatmatthew) [reply FoxHound.Web/Controllers/CommentController.cs:27]: Fixed.

**Pull request merged**